### PR TITLE
Fix #785 JacksonXML support

### LINKED ISSUE #785: JacksonXML support
This might work already, we are perhaps just missing a test.

### PR DESCRIPTION
**Pull request by ppalaga**
Fix #785 JacksonXML support

### CODE REVIEW

**Comment by ppalaga** [comment]
f45e450 added the missing `<module>jackson-dataformat-xml</module>` in the support parent.

File: extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        Stream.concat(
+                Stream.of(
+                        JsonFactory.class,
+                        ObjectCodec.class,
+                        VerifierFactoryLoader.class,
+                        DatatypeLibraryFactory.class,
+                        XMLEventFactory.class,
+                        XMLInputFactory.class,
+                        XMLOutputFactory.class)
+                        .map(Class::getName),
```
Comment (lburgazzoli) [inline extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java]: maybe we should just use class names as strings

File: extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                Stream.of(
+                        XMLValidationSchemaFactory.INTERNAL_ID_SCHEMA_DTD,
+                        XMLValidationSchemaFactory.INTERNAL_ID_SCHEMA_RELAXNG,
+                        XMLValidationSchemaFactory.INTERNAL_ID_SCHEMA_W3C,
+                        XMLValidationSchemaFactory.INTERNAL_ID_SCHEMA_TREX)
+                        .map(schemaId -> XMLValidationSchemaFactory.class.getName() + "." + schemaId))
```
Comment (lburgazzoli) [inline extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java]: As there's no discovery but we deal with a fixed set of ites, maybe we can use strings so we can avoid the the two `Stream.of` and the `Stream.concat`

File: extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java (hunk 1 above)
Comment (ppalaga) [reply extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java]: Yeah, we discussed this some time ago with the result that strings are faster and they do not have any disadvantages related to class loading. In the meantime, I asked David about this and he said he finds the perf drawbacks negligible and so I started preferring maintainability. WDYT?

File: extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java (hunk 2 above)
Comment (ppalaga) [reply extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java]: Yes, your proposal is valid and I can see the advantages but classes are easier to type in an IDE and any typos will cause a compilation error so it feels safer. I am open to discuss this.

Comment (lburgazzoli) [reply extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java]: Yes it is not really related to perf of loading classes but more about about the `complexity` of two nested streams with mapping like looking at the code is seems much more complex that what it actually does but there's no problem to keep it as it is

File: extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java (hunk 1 above)
Comment (lburgazzoli) [reply extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java]: I'm always +1 for maintainability you know :)

File: extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java (hunk 2 above)
Comment (ppalaga) [reply extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java]: I am at least hanging this quote from Plato's Timaeus above my table to remind me not to sin against the beauty in the future ;)

> The good, of course, is always beautiful, and the beautiful never lacks proportion.

**Pull request merged**